framework/src/org/checkerframework/framework/util/defaults/QualifierDefaults.java

```diff
@@ -525,10 +526,13 @@ private void applyDefaultsElement(final Element annotationScope, final Annotated
                 // but it doesn't work correctly and tests fail.
                 // (That whole @FromStubFile and @FromByteCode annotation
                 // logic should be replaced by something sensible.)
-            if ((ElementUtils.isElementFromByteCode(annotationScope) &&
+        	SourceChecker checker = atypeFactory.getContext().getChecker();
+        	if ((ElementUtils.isElementFromByteCode(annotationScope) &&
                     atypeFactory.declarationFromElement(annotationScope) == null &&
-                    !atypeFactory.isFromStubFile(annotationScope)) ||
-                    !annotatedForThisChecker
+                    !atypeFactory.isFromStubFile(annotationScope) &&
+                    checker.hasOption("safeDefaultsForUnannotatedBytecode")) || // TODO: make safeDefaultsForUnannotatedBytecode the default behavior.
```

Comment:
The code will change only because the -AsafeDefaultsForUnannotatedBytecode option will be removed and replaced with its opposite -AunsafeDefaultsForUnannotatedBytecode. But the TODO isn't really needed since this code will be found when doing a search for all instances of "safeDefaultsForUnannotatedBytecode".